specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,305 @@
+
+# Description
+
+This event will be raised before the external protocol launch occurs. Currently a popup dialog is displayed in which the user can click `Open` or `Cancel`. If the request is made from a [trustworthy origin](#https://w3c.github.io/webappsec-secure-contexts/#potentially-trustworthy-origin) a checkmark box will be displayed that will allow the user to always allow this external protocol from this origin. The `NavigationStarting`, `NavigationCompleted`, `SourceChanged`, `ContentLoading`, and `HistoryChanged` events will not be raised when a request is made to launch an external protocol.
+
+The `LaunchingExternalProtocol` event will be raised on either `CoreWebView2` or `CoreWebView2Frame` depending on if the launch request is originating from the main frame or a non-main frame. A current limitation of our `CoreWebView2Frame` API is that it only supports top level iframes. Any nested iframes will not have a `CoreWebView2Frame` associated with them. In the case of a nested iframe requesting the external protocol launch, the event will be raised from the top level iframe.
+# Examples
+
+## Win32 C++
+    
+```cpp 
+
+AppWindow* m_appWindow;
+wil::com_ptr<ICoreWebView2> m_webView;
+EventRegistrationToken m_launchingExternalProtocolToken = {};
+EventRegistrationToken m_frameLaunchingExternalProtocolToken = {};
+EventRegistrationToken m_frameCreatedToken = {};
+
+void RegisterLaunchingExternalProtocolHandler()
+{
+    auto webView5 = m_webView.try_query<ICoreWebView2_5>();
+    if (webView5)
+    {
+        CHECK_FAILURE(webView5->add_LaunchingExternalProtocol(
+            Callback<ICoreWebView2LaunchingExternalProtocolEventHandler>(
+                [this](
+                    ICoreWebView2* sender,
+                    ICoreWebView2LaunchingExternalProtocolEventArgs* args) {
+                        wil::unique_cotaskmem_string uri;
+                        CHECK_FAILURE(args->get_Uri(&uri));
+                        if (wcsicmp(uri.get(), L"calculator://") == 0)
+                        {
+                            CHECK_FAILURE(args->put_Handled(TRUE));
+                            // If this matches our desired protocol, then suppress the
+                            // popup dialog.
+                        }
+                        else
+                        {
+                            // Otherwise allow the popup dialog, and allow the user to decide
+                            // whether or not to allow the protocol to launch.
+                        }
+                    return S_OK;
+                })
+                .Get(),
+            &m_launchingExternalProtocolToken));
+    }
+    auto webView4 = m_webview.try_query<ICoreWebView2_4>();
```

Comment:
If you have ICoreWebView2_5 you don't need to QI for ICoreWebView2_4 since _5 derives from _4.